### PR TITLE
Create footer

### PR DESCRIPTION
**Pull request by igorstalmach**
- working icons and mailto link
   - changes opacity and pointer on hover
- responsive layout (mobile & desktop view)
- working nav links (same objects as main nav)

also:
- edited IconProps and icon files to support variable sizes and colors

### CODE REVIEW

File: src/icons/InstagramIcon.tsx [hunk 1]
```diff
@@ -2,12 +2,17 @@ import React from 'react';
...
-            <path d="M16.5814 19.876C14.898 19.876 13.5288 18.507 13.5288 16.8247C13.5288 15.1424 14.898 13.7734 16.5814 13.7734C18.264 13.7734 19.6322 15.1424 19.6322 16.8247C19.6322 18.507 18.264 19.876 16.5814 19.876ZM16.5814 12.0613C13.9541 12.0613 11.8164 14.1977 11.8164 16.8247C11.8164 19.4517 13.9541 21.5881 16.5814 21.5881C19.2079 21.5881 21.3456 19.4517 21.3456 16.8247C21.3456 14.1977 19.2079 12.0613 16.5814 12.0613Z" fill="white"/>
-            <path d="M12.6377 8.9724C10.4375 8.9724 8.64774 10.762 8.64774 12.9628V20.6861C8.64774 22.886 10.4375 24.6764 12.6377 24.6764H20.3622C22.5633 24.6764 24.354 22.886 24.354 20.6861V12.9628C24.354 10.762 22.5633 8.9724 20.3622 8.9724H12.6377ZM20.3622 26.4958H12.6377C9.43396 26.4958 6.82715 23.8894 6.82715 20.6861V12.9628C6.82715 9.75947 9.43396 7.153 12.6377 7.153H20.3622C23.5659 7.153 26.1736 9.75947 26.1736 12.9628V20.6861C26.1736 23.8894 23.5659 26.4958 20.3622 26.4958Z" fill="white"/>
+        <svg width={width} height={height} viewBox="0 0 33 34" fill="none" xmlns="http://www.w3.org/2000/svg" onClick={() => window.location.href = 'https://www.instagram.com/poliwrocket/'}>
```
Comment (JanZmyslowski) [inline src/icons/InstagramIcon.tsx:11]: same as above

File: src/icons/TikTokIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { IconProps } from '../Contract';
+
+export default function TikTokIcon(props: IconProps) {
+    const width = props.width ?? 33;
+    const height = props.height ?? 33;
+    const color = props.color ?? 'white';
+    const secondaryColor = props.secondaryColor ?? 'black';
+
+    return (
+        <svg width={width} height={height} viewBox="0 0 33 33" fill="none" xmlns="http://www.w3.org/2000/svg" onClick={() => window.location.href = 'https://www.tiktok.com/@poliwrocket'}>
```
Comment (JanZmyslowski) [inline src/icons/TikTokIcon.tsx:11]: same as above

File: src/icons/YouTubeIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { IconProps } from '../Contract';
+
+export default function YouTubeIcon(props: IconProps) {
+    const width = props.width ?? 33;
+    const height = props.height ?? 33;
+    const color = props.color ?? 'white';
+    const secondaryColor = props.secondaryColor ?? 'black';
+
+    return (
+        <svg width={width} height={height} viewBox="0 0 33 33" fill="none" xmlns="http://www.w3.org/2000/svg" onClick={() => window.location.href = 'https://www.youtube.com/@poliwrocket1294'}>
```
Comment (JanZmyslowski) [inline src/icons/YouTubeIcon.tsx:11]: same as above

File: src/Contract.tsx [hunk 4]
```diff
@@ -4,7 +4,9 @@ export interface NavMenuItem {
 }
 
 export interface IconProps {
-    size?: number;
+    height?: number;
+    width?: number;
```
Comment (JanZmyslowski) [inline src/Contract.tsx:8]: Majority of icons are square/circle and making it possible for them to change proportions is not a good approach, the same thing to color.
In my option, we should introduce some generic type of hierarchy to handle it, e.g.:
image

interface BaseIconProps<T> {
    size?: T;
    color?: string;
    onClick?: () => void;
}

interface IconSize {
    width: number;
    height: number;
}

export interface IconProps extends BaseIconProps<number> {
    size?: number;
    color?: string;
    onClick?: () => void;
}

export interface AdvancedIconProps extends BaseIconProps<IconSize> {
    secondaryColor?: string;
}

**Review by JanZmyslowski — CHANGES_REQUESTED** [review]

File: src/Contract.tsx [hunk 5]
```diff
@@ -3,10 +3,30 @@ export interface NavMenuItem {
...
+export interface IconProps extends BaseIconProps<number, string> {}
+
+// Icons that only require one size value AND a secondary color.
+export interface IconPropsWithColor extends BaseIconProps<number, IconColor> {}
+
+// Icons that require both width and height.
+export interface AdvancedIconProps extends BaseIconProps<IconSize, string> {}
+
+// Icons that require both width and height AND a secondary color.
+export interface AdvancedIconPropsWithColor extends BaseIconProps<IconSize, IconColor> {}
```
Comment (igorstalmach) [inline src/Contract.tsx]: I created four different interfaces covering different use cases

File: src/icons/YouTubeIcon.tsx [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
 
-export default function YouTubeIcon(props: IconProps) {
-    const width = props.width ?? 33;
-    const height = props.height ?? 33;
-    const color = props.color ?? 'white';
-    const secondaryColor = props.secondaryColor ?? 'black';
+export default function YouTubeIcon(props: IconPropsWithColor) {
+    const size = props.size ?? 33;
+    const color = props.color?.primaryColor ?? 'white';
+    const secondaryColor = props.color?.secondaryColor ?? 'black';
```
Comment (igorstalmach) [inline src/icons/YouTubeIcon.tsx:7]: Also I thought that values of attributes of each icon component (e.g. size, color) after ?? operator can be used as default values - so size and color props are only used when creating icon different than default.

File: src/components/navbar/NavBar.tsx [hunk 7]
```diff
@@ -18,11 +18,11 @@ export default function NavBar(props: ISelfProps) {
...
         <div className={styles.headerContainer}>
             <div className={styles.headerWrapper}>
-                <IconLogo height={42} width={116} onClick={() => window.location.href = window.appContext.baseUrl}/>
+                <IconLogo onClick={() => window.location.href = window.appContext.baseUrl}/>
                 <div className={classes(styles.linksWrapper, !menuOpen && styles.linksHidden)}>
                     <div className={styles.menu} onClick={() => setMenuOpen(!menuOpen)}>
-                        {!menuOpen && <IconMenu height={42} width={42} color={'white'}/>}
-                        {menuOpen && <IconClose height={42} width={42} color={'white'}/>}
+                        {!menuOpen && <IconMenu/>}
+                        {menuOpen && <IconClose/>}
```
Comment (igorstalmach) [inline src/components/navbar/NavBar.tsx:25]: ad. to the comment above. This approach can help in writing less code with less UI bugs.

File: src/icons/YouTubeIcon.tsx (hunk 6 above)
Comment (JanZmyslowski) [reply src/icons/YouTubeIcon.tsx:7]: Yes, nullable operator is a good approach for default values

File: src/components/navbar/NavBar.tsx (hunk 7 above)
Comment (JanZmyslowski) [reply src/components/navbar/NavBar.tsx:25]: That`s correct

**Review by JanZmyslowski — APPROVED** [review]

**Pull request merged**